rmw_zenoh_cpp/src/detail/rmw_publisher_data.cpp

```diff
@@ -107,40 +107,36 @@ std::shared_ptr<PublisherData> PublisherData::make(
     return nullptr;
   }
 
-  zenoh::ZResult result;
-  std::optional<zenoh::ext::PublicationCache> pub_cache;
-  zenoh::KeyExpr pub_ke(entity->topic_info()->topic_keyexpr_);
+  auto adv_pub_opts = zenoh::ext::SessionExt::AdvancedPublisherOptions::create_default();
+  adv_pub_opts.publisher_detection = true;
+  adv_pub_opts.sample_miss_detection = true;
+
   // Create a Publication Cache if durability is transient_local.
   if (adapted_qos_profile.durability == RMW_QOS_POLICY_DURABILITY_TRANSIENT_LOCAL) {
-    zenoh::ext::SessionExt::PublicationCacheOptions pub_cache_opts =
-      zenoh::ext::SessionExt::PublicationCacheOptions::create_default();
-
-    pub_cache_opts.history = adapted_qos_profile.depth;
-    pub_cache_opts.queryable_complete = true;
-
-    std::string queryable_prefix = entity->zid();
-    pub_cache_opts.queryable_prefix = zenoh::KeyExpr(queryable_prefix);
-
-    pub_cache = session->ext().declare_publication_cache(
-      pub_ke, std::move(pub_cache_opts), &result);
```

Comment:
We previously needed to do this to fix https://github.com/ros2/rmw_zenoh/issues/263. Can you confirm that we no longer need to explicitly call `QueryingSubscriber::get()` when we discover an AdvancedPublisher in the graph?

Reply:
We should no longer need this hack. Otherwise, it's a bug in zenoh.